<launch>
    <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

    <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
    <node pkg="nodelet" type="nodelet" name="pointcloud_downsize" args="load pcl/VoxelGrid pcl_manager" output="screen">
        <remap from="~input" to="/zed2/zed_node/point_cloud/cloud_registered" />
        <rosparam>
            filter_field_name: z
            filter_limit_min: 0
            filter_limit_max: 0
            filter_limit_negative: True
            leaf_size: 0.09
        </rosparam>
    </node>
</launch>